frontend/src/metabase/components/ErrorPages/use-error-info.ts

```diff
@@ -0,0 +1,120 @@
+    const matches = location.match(
+      /(question|model|dashboard|collection)[[\/\#]([\d\w]+)/,
+    );
+
+    const entity = (matches?.[1] ?? undefined) as
+      | ReportableEntityName
+      | undefined;
+    const id = matches?.[2] ?? "";
+
+    const isAdHoc = entity === "question" && window.location.href.includes("#");
+
+    const entityInfoRequest = getEntityDetails({ entity, id, isAdHoc });
+    const bugReportDetailsRequest = isAdmin
+      ? UtilApi.bug_report_details().catch(nullOnCatch)
+      : Promise.resolve(null);
+
+    const sessionPropertiesRequest = SessionApi.properties().catch(nullOnCatch);
+
+    const logsRequest: any = isAdmin
+      ? UtilApi.logs().catch(nullOnCatch)
+      : Promise.resolve(null);
+
+    /* eslint-disable no-console */
+    // @ts-expect-error I'm sorry
+    const frontendErrors = console?.errorBuffer?.map?.(err => err.join(""));
+    /* eslint-enable no-console */
+
+    const settledPromises = await Promise.allSettled([
+      entityInfoRequest,
+      bugReportDetailsRequest,
+      sessionPropertiesRequest,
+      logsRequest,
+    ]);
+
+    const [entityInfo, bugReportDetails, sessionProperties, logs] =
+      settledPromises.map((promise: any) => promise.value);
+
+    const queryResults =
+      hasQueryData(entity) &&
+      entityInfo?.dataset_query &&
+      (await MetabaseApi.dataset(entityInfo.dataset_query).catch(nullOnCatch));
+
+    // if this is an ad-hoc exploration on top of a saved question, fetch the original card
+    if (hasQueryData(entity) && entityInfo?.original_card_id) {
+      entityInfo.originalCard = await getEntityDetails({
+        entity,
+        id: entityInfo.original_card_id,
+      });
+    }
+
+    const filteredLogs = logs?.slice?.(0, 100);
+    const backendErrors = logs?.filter?.((log: any) => log.level === "ERROR");
+
+    const userLogs = logs?.filter(
+      (log: any) =>
+        log?.msg?.includes?.(`{:metabase-user-id ${currentUser.id}}`) ||
+        log?.msg?.includes?.(` userID: ${currentUser.id} `),
+    );
+
+    const { engines, ...sessionPropertiesWithoutEngines } = sessionProperties;
+
+    const payload: ErrorPayload = {
+      url: location,
+      entityInfo,
+      entityName: entity,
+      localizedEntityName: getLocalizedEntityName(entity),
+      ...(queryResults ? { queryResults } : undefined),
+      logs: filteredLogs,
+      frontendErrors,
+      backendErrors,
+      userLogs,
+      instanceInfo: {
+        sessionProperties: sessionPropertiesWithoutEngines,
+        bugReportDetails,
+      },
+    };
+
+    return payload;
+  }, [enabled]);
+};
+
+const nullOnCatch = () => null;
+
+const getLocalizedEntityName = (entityName?: ReportableEntityName) => {
+  switch (entityName) {
+    case "question":
+      return t`question`;
+    case "model":
+      return t`model`;
+    case "dashboard":
+      return t`dashboard`;
+    case "collection":
+      return t`collection`;
+    default:
+      return entityName;
```

Comment:
The argument can be undefined, in which case this will return undefined, which might for all I know lead to a runtime error, I think, or the use of the string 'undefined'

Reply:
if `entityName` is undefined, we want `localizedEntityName` to also be undefined - we check the truthiness of `entityName` to determine whether or not to show `localizedEntityName`